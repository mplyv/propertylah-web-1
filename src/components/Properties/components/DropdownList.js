import styles from '../property.module.css'

const DropdownList = (props)=>{
    return (
        <select id={props.id} className={styles[props.className]} onChange={props.onChange} value={props.value}>
            {props.options.map((selection)=>{
                return(
                    <option key={selection}>{selection}</option>
                )
            })}
        </select>
    )
}

export default DropdownList

// props = {id, className, onChange, value, options}
// options = options to appear on dropdown list
// id to handle if case for onChange function.
// className for styling

// see PopupComponent.js or FilterBar.js for example