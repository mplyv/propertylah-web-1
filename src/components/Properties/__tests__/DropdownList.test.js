import {render, screen, cleanup, fireEvent} from '@testing-library/react';
import renderer from 'react-test-renderer'
import DropdownList from '../components/DropdownList'
import userEvent from '@testing-library/user-event'


//* cleanup after each test
afterEach(() => {
  cleanup();
});

let propertyOptions = ['Property Type: All','HDB','Condominium','Apartment']
let propertyType = 'All'
const onChange = jest.fn();

test('Everything should render',()=>{
    render(<DropdownList onChange={onChange} value={propertyType} options={propertyOptions} className={`filterSearchBarButton`}/>);
    expect(screen.getByTestId('dropdownListTest')).toBeInTheDocument();
    expect(screen.getByTestId('test-HDB')).toBeInTheDocument();
    expect(screen.getByTestId('test-Apartment')).toBeInTheDocument();
    expect(screen.getByTestId('test-Condominium')).toBeInTheDocument();
});

//* different test method because this one using a select element. using user.click doesn't trigger the function
test('Changing selection triggers onChange function',async ()=>{
    const {rerender} = render(<DropdownList onChange={onChange} value={propertyType} options={propertyOptions} className={`filterSearchBarButton`}/>);
    fireEvent.change(screen.getByTestId('dropdownListTest'), {
    target: { value: "HDB" },});
    fireEvent.change(screen.getByTestId('dropdownListTest'), {
    target: { value: "Apartment" },});
    fireEvent.change(screen.getByTestId('dropdownListTest'), {
    target: { value: "Condominium" },});
    expect(onChange).toHaveBeenCalled();
});


// Snapshot tests
test('match propertyType snapshot for DropdownList',()=>{
    const tree = renderer.create(<DropdownList onChange={onChange} value={propertyType} options={propertyOptions} className={`filterSearchBarButton`}/>).toJSON();
    expect(tree).toMatchSnapshot();
})
