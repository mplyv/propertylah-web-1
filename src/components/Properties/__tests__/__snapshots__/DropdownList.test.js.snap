// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`match propertyType snapshot for DropdownList 1`] = `
<select
  className="filterSearchBarButton"
  data-testid="dropdownListTest"
  onChange={[MockFunction]}
  value="All"
>
  <option
    data-testid="test-Property Type: All"
  >
    Property Type: All
  </option>
  <option
    data-testid="test-HDB"
  >
    HDB
  </option>
  <option
    data-testid="test-Condominium"
  >
    Condominium
  </option>
  <option
    data-testid="test-Apartment"
  >
    Apartment
  </option>
</select>
`;
